src/helper/selection-tools/rotate-tool.js

```diff
@@ -44,13 +41,9 @@ class RotateTool {
             
             if (event.modifiers.shift) {
                 rotAngle = Math.round(rotAngle / 45) * 45;
-                item.applyMatrix = false;
-                item.pivot = this.rotGroupPivot;
-                item.rotation = rotAngle;
-            } else {
-                item.rotate(rotAngle - this.prevRot[i], this.rotGroupPivot);
             }
-            this.prevRot[i] = rotAngle;
+            item.rotate(rotAngle - this.prevRot, this.rotGroupPivot);
+            this.prevRot = rotAngle;
```

Comment:
I'm seeing an issue where you can only rotate one thing
![rotbug](https://user-images.githubusercontent.com/2855464/38319968-40c6c11e-3801-11e8-8538-670fd7ce7d8f.gif)

I think you want to bring this line out of the loop